python/py-setuptools-rust/Portfile

```diff
@@ -27,11 +28,9 @@ if {${name} ne ${subport}} {
                     port:py${python.version}-setuptools_scm \
                     port:py${python.version}-wheel
```

Comment:
since the package supports it we should transition to build according to PEP517 (e.g., here `pip` is not needed as dependency). 

I cannot leave comments in parts of the file that has not been changed, so I'll describe my suggested changes here:

under the `python.versions` line add:
```python.pep517 yes```, this will automatically add `setuptools` and `wheel` to `depends_build`; the only other dependency that needs to stay under `depends-build` is `setuptools_scm`